.github/workflows/rust.yml

```diff
@@ -84,9 +84,13 @@ jobs:
     name: Nix-Build Tests
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: cachix/install-nix-action@v15
+      - uses: actions/checkout@v2.4.0
+      - uses: cachix/install-nix-action@v16
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+      - uses: cachix/cachix-action@v10
         with:
-          nix_path: nixpkgs=channel:nixos-21.11
-      - name: Running nix-build
-        run: nix-build default.nix
+          name: mini
+          signingKey: '${{ secrets.CACHIX_SIGNING_KEY }}'
+      - run: nix-build default.nix
+      - run: nix-shell --run "echo OK"
```

Comment:
Good idea to test the shell! Was wondering for a moment what it is for though, maybe you could echo "building nix-shell works"?